### ISSUE TITLE
Unexpected error when executing non-transaction point get

### ISSUE THREAD

**Issue by pcqz** [pingcap/tidb#45662]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> select k from sbtest1 where id=(select id from sbtest2 where id=100);
+------+
| k    |
+------+
| 4996 |
+------+
1 row in set (0.00 sec)

mysql> commit;
Query OK, 0 rows affected (0.00 sec)

mysql> select k from sbtest1 where id=(select id from sbtest2 where id=101);
+------+
| k    |
+------+
| 5035 |
+------+
1 row in set (0.00 sec)

mysql> desc select k from sbtest1 where id=(select id from sbtest2 where id=101);
+--------------------+---------+------+---------------+-------------------+
| id                 | estRows | task | access object | operator info     |
+--------------------+---------+------+---------------+-------------------+
| Projection_11      | 1.00    | root |               | sbtest1.sbtest1.k |
| └─Point_Get_12     | 1.00    | root | table:sbtest1 | handle:101        |
+--------------------+---------+------+---------------+-------------------+
2 rows in set (0.01 sec)

```

### 2. What did you expect to see? (Required)

No error log seen in TiDB log.

### 3. What did you see instead (Required)

[2023/07/28 17:32:29.638 +08:00] [ERROR] [optimistic.go:139] ["failed init txnStartTS with MaxUint64"] [error="cannot force prepare const start ts because txn is active"] [conn=4450607564958204317] [text="select k from sbtest1 where id=(select id from sbtest2 where id=101)"]

### 4. What is your TiDB version? (Required)
v6.5.1

**Comment by lcwangchao** [comment]
This issue has been fixed by: #40197 though this PR is not intended to do this...

**Comment by lcwangchao** [comment]
#40197 has been cherry-pick to release-6.5: https://github.com/pingcap/tidb/pull/44092
